pkg/dataobj/index/calculate.go

```diff
@@ -160,22 +172,25 @@ func (c *Calculator) processLogsSection(ctx context.Context, sectionLogger log.L
 		return fmt.Errorf("failed to read log section stats: %w", err)
 	}
 
-	columnBloomBuilders := make(map[string]*bloom.BloomFilter)
-	columnIndexes := make(map[string]int64)
-	for _, column := range stats.Columns {
-		logsType, _ := logs.ParseColumnType(column.Type)
-		if logsType != logs.ColumnTypeMetadata {
-			continue
+	calculationContext := &logsCalculationContext{
+		tenantID:       tenantID,
+		objectPath:     objectPath,
+		sectionIdx:     sectionIdx,
+		streamIDLookup: streamIDLookup,
+		builderMtx:     c.builderMtx,
```

Comment:
nit: sharing mutex pointers can be a source of bugs because the spread of where logic is defined gets wider and it can be harder to keep track of making sure the mutex is being used properly in all places. 

I'd recommend considering changing the semantics such that:

* The mutex is locked whenever we pass the logsCalculationContext to ProcessBatch/Flush 
* We clarify in the interface API that implementations must not retain the builder after the call returns